### ISSUE TITLE
`missing_docs_in_private_items` warns for `const _`

### ISSUE THREAD

**Issue by clarfonthey** [rust-lang/rust-clippy#13427]
### Summary

This item cannot have generated docs, so, shouldn't really be documented.

### Lint Name

`missing_docs_in_private_items`

### Reproducer

I tried this code:

```rust
#![warn(clippy::missing_docs_in_private_items)]
const _: () = {
     assert!(1 == 1);
};
```

I saw this happen:

```
warning: missing documentation for a constant
 --> src/lib.rs:2:1
  |
2 | / const _: () = {
3 | |     assert!(1 == 1);
4 | | };
  | |__^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#missing_docs_in_private_items
```

When this shouldn't trigger at all.

### Version

rustc 1.83.0-nightly (506f22b46 2024-09-19)
binary: rustc
commit-hash: 506f22b4663f3e756e1e6a4f66c6309fdc00819c
commit-date: 2024-09-19
host: x86_64-unknown-linux-gnu
release: 1.83.0-nightly
LLVM version: 19.1.0
